pkg/services/libraryelements/database.go

```diff
@@ -267,6 +285,24 @@ func getLibraryElements(c context.Context, store db.DB, cfg *setting.Cfg, signed
 
 	leDtos := make([]model.LibraryElementDTO, len(libraryElements))
 	for i, libraryElement := range libraryElements {
+		var modelMap map[string]interface{}
+
+		err := json.Unmarshal(libraryElement.Model, &modelMap)
+		if err != nil {
+			return []model.LibraryElementDTO{}, err
+		}
+
+		if libraryPanel, ok := modelMap["libraryPanel"].(map[string]interface{}); ok {
+			if uid, ok := libraryPanel["uid"]; ok && uid == "" {
+				libraryPanel["uid"] = libraryElement.UID
+			}
+		}
+
+		updatedModel, err := json.Marshal(modelMap)
+		if err != nil {
+			return []model.LibraryElementDTO{}, err
+		}
+
```

Comment:
if the LibraryPanel doesn't have an id, I set it before returning the model

Reply:
I would move this logic outside of the database layer, to the service.